app/bundles/FormBundle/Tests/BooleanFieldTest.php

```diff
@@ -0,0 +1,316 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\FormBundle\Tests;
+
+use Mautic\CoreBundle\Test\MauticMysqlTestCase;
+use Mautic\FormBundle\Entity\Field;
+use Mautic\FormBundle\Model\SubmissionModel;
+use Twig\Environment;
+
+class BooleanFieldTest extends MauticMysqlTestCase
+{
+    protected SubmissionModel $submissionModel;
+    protected \ReflectionMethod $normalizeValueMethod;
+    protected Environment $twig;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->submissionModel      = static::getContainer()->get('mautic.form.model.submission');
+        $reflection                 = new \ReflectionClass($this->submissionModel);
+        $this->normalizeValueMethod = $reflection->getMethod('normalizeValue');
+        $this->normalizeValueMethod->setAccessible(true);
+        $this->twig = static::getContainer()->get('twig');
```

Comment:
I'd suggest to create the form with the field, save it to the database and access the form's public page and submit it. Such test would ensure the feature is and continue working in the future. Rather than messing with reflections. This test mis making shortcuts that can hide a bug. And it is actually more complicated to write.